flowToScript.js

```diff
@@ -1,55 +1,126 @@
 //Build a js script based on the flow. How hard could it be?
-
+let requestString;
+let makeRequestString;
 function createScript(chartProperties) {
   //Start with root block
   //Kicks off depth-first tree traversal
   chartProperties = chartProperties;
-  evaluateBlock(0);
+  requestString = 'let resp; \n let requestBlock = async () => { \n';
+
+  if (baseUrl != '') {
+    makeRequestString = `let makeRequest = async (baseUrl, method, path, data) => {
+    const https = require("node:${baseUrl.substring(
+      0,
+      baseUrl.lastIndexOf(':')
+    )}");
+    let response;
+    const options = {
+      hostname: baseUrl,
+      path: path,
+      method: method,
+      data: data,
+      headers: {
+        'api-key': null
+      }
+    };
+  
+    const req = https.request(options, (res) => {
+      console.log("statusCode:", res.statusCode);
+      console.log("headers:", res.headers);
+      response = res;
+      res.on("data", (data) => {
+        process.stdout.write(data);
+      });
+    });
+  
+    req.on("error", (e) => {
+      console.error(e);
+    });
+    req.end();
+    return response;
+    };
+    requestBlock();`;
+    if (document.getElementById('apiKey').value) {
```

Comment:
easier to do `'api-key': ${document.getElementById('apiKey').value || 'null'}` on line 23